src/node/node_state.h

```diff
@@ -426,9 +425,44 @@ namespace ccf
             setup_consensus(resp.network_info.public_only);
             setup_history();
 
+            if (!config.joining.snapshot.empty())
+            {
+              // It is only possible to deserialise the snapshot then, once the
+              // ledger secrets have been passed in by the network
+              LOG_DEBUG_FMT(
+                "Deserialising snapshot ({})", config.joining.snapshot.size());
+              auto rc =
+                network.tables->deserialise_snapshot(config.joining.snapshot);
+
+              if (rc != kv::DeserialiseSuccess::PASS)
+              {
+                throw std::logic_error(
+                  fmt::format("Failed to apply snapshot: {}", rc));
+              }
+
+              kv::ReadOnlyTx tx;
+              auto sig_view = tx.get_read_only_view(network.signatures);
+              auto sig = sig_view->get(0);
+              if (!sig.has_value())
+              {
+                throw std::logic_error(
+                  fmt::format("No signatures found after applying snapshot"));
+              }
+
+              auto seqno = network.tables->current_version();
+              consensus->force_become_backup(seqno, sig->view);
+
+              reset_data(config.joining.snapshot);
```

Comment:
Once the snapshot has been applied, we don't want to keep it around in enclave memory.